test/manipulation.test.js

```diff
@@ -215,6 +342,29 @@ describe('manipulation', function () {
           });
         });
     });
+
+    it('should preserve properties with "undefined" value (promise variant)', function(done) {
+      Person.create({ name: 'a-name', gender: undefined })
+      .then(function (created) {
+        created.toObject().should.have.properties({
+          id: created.id,
+          name: 'a-name',
+          gender: undefined
+        });
+
+        return Person.findById(created.id)
+        .then(function(found) {
+          found.toObject().should.have.properties({
+            id: created.id,
+            name: 'a-name',
+            gender: undefined
+          });
+          done();
+        });
+      }).catch(done);
+
+    });
+
```

Comment:
Same here, "preserve properties with undefined value" is not related to callback vs. promise, can we omit this test please?